### PR TITLE
fix: math reward fn

### PR DESCRIPTION
**Pull request by yueyang130**
Currently, the reward fn intends to give  +0.1 reward to encourage output answer in the format of `\boxed{}`.

However, in `CustomRewardManager` class, it uses `sequences = torch.cat((valid_prompt_ids, valid_response_ids))` as the input of `self.compute_score`. In this way,  var `seqences` includes the system prompt "Please reason step by step, and put your final answer within \boxed{}.". 

Note that the system prompt already includes `\boxed{}`. Thus the reward fn gives +0.1 even the model acutally output without \boxed{}. An example is below:

Fix this bug by passing the response without system prompt into `self.compute_score`

image

### CODE REVIEW

**Review by hiyouga — APPROVED** [review]
Nice catch!
